datasetinsights/estimators/faster_rcnn.py

```diff
@@ -22,6 +22,594 @@
+        model_name = f"fasterrcnn_{self.config.backbone}_fpn"
+        self.model = torchvision.models.detection.__dict__[model_name](
+            num_classes=config.num_classes,
+            pretrained_backbone=config.pretrained_backbone,
+            pretrained=config.pretrained,
+            box_detections_per_img=MAX_BOXES_PER_IMAGE,
+            box_score_thresh=box_score_thresh,
+        )
+        self.model_without_ddp = self.model
+        self.gpu = gpu
+        self.rank = rank
+        self.sync_metrics = config.get("synchronize_metrics", True)
+        logger.info(f"gpu: {self.gpu}, rank: {self.rank}")
+
+        self.checkpointer = checkpointer
+        checkpoint_file = config.get("checkpoint_file", const.NULL_STRING)
+        if checkpoint_file != const.NULL_STRING:
+            checkpointer.load(self, config.checkpoint_file)
+
+        self.metrics = {}
+        for metric_key, metric in config.metrics.items():
+            self.metrics[metric_key] = EvaluationMetric.create(
+                metric.name, **metric.args
+            )
+        self.model.to(self.device)
+
+        if self.config.system.distributed:
+            self.model = torch.nn.parallel.DistributedDataParallel(
+                self.model, device_ids=[self.gpu]
+            )
+            self.model_without_ddp = self.model.module
+
+    def train(self, **kwargs):
+        config = self.config
+        train_dataset = download_data(config, TRAIN)
+        val_dataset = download_data(config, VAL)
+        label_mappings = train_dataset.label_mappings
+        if self.config.system.dryrun:
+            train_dataset = create_dryrun_dataset(config, train_dataset, TRAIN)
+            val_dataset = create_dryrun_dataset(config, val_dataset, VAL)
+
+        logger.info(f"length of train dataset is {len(train_dataset)}")
+        logger.info(f"length of validation dataset is {len(val_dataset)}")
+        is_distributed = config.system.distributed
+        train_sampler = create_sampler(is_distributed=is_distributed,
+                                       dataset=train_dataset,
+                                       is_train=True)
+        val_sampler = create_sampler(is_distributed=is_distributed,
+                                     dataset=val_dataset,
+                                     is_train=False)
+
+        train_loader = create_dataloader(config, train_dataset, train_sampler,
+                                         TRAIN)
+        val_loader = create_dataloader(config, val_dataset, val_sampler,
+                                       VAL)
+        self.train_loop(
+            train_dataloader=train_loader,
+            label_mappings=label_mappings,
+            val_dataloader=val_loader,
+            train_sampler=train_sampler,
+        )
+
+    def train_loop(
+        self,
+        *,
+        train_dataloader,
+        label_mappings,
+        val_dataloader,
+        train_sampler=None,
+    ):
+        """
+        Args:
```

Comment:
The docstring should be improved here if this is a public method. Alternatively, you can make this private method and be more relaxed on the docstring.

Reply:
Fixed